app/models/component.rb

```diff
@@ -75,4 +74,24 @@ def type
   def display
     I18n.t("user_journey.component_name.#{type}")
   end
+
+private
+
+  def unexpired_encryption_certificate
+    x509 = encryption_certificate&.x509
+    if x509.nil? || x509.not_after < Time.now
+      Rails.logger.error "When publishing the meta data the service '#{name}' has been identified as having an expired encryption certificate."
+      nil
+    else
+      encryption_certificate
+    end
+  end
+
+  def unexpired_enabled_signing_certificates
+    valid_certs = enabled_signing_certificates.reject { |cert| cert.x509.not_after < Time.now }
```

Comment:
Gonna approve, but I like The O Father's suggestion of having a `expired?` method on the certificate objects. Will defo be useful in the future